thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/util/ThirdEyeUtils.java

```diff
@@ -370,9 +372,19 @@ public static DatasetConfigDTO getDatasetConfigFromName(String dataset) {
     return datasetConfig;
   }
 
-  public static String getDatasetFromMetricFunction(MetricFunction metricFunction) {
-    MetricConfigDTO metricConfig = getMetricConfigFromId(metricFunction.getMetricId());
-    return metricConfig.getDataset();
+  public static List<DatasetConfigDTO> getDatasetConfigsFromMetricUrn(String metricUrn) {
+    DatasetConfigManager datasetConfigManager = DAORegistry.getInstance().getDatasetConfigDAO();
+    MetricEntity me = MetricEntity.fromURN(metricUrn);
+    MetricConfigDTO metricConfig = DAORegistry.getInstance().getMetricConfigDAO().findById(me.getId());
+    if (metricConfig == null) return new ArrayList<>();
+    if (!metricConfig.isDerived()) {
+      return Collections.singletonList(datasetConfigManager.findByDataset(metricConfig.getDataset()));
+    } else {
```

Comment:
no need else

Reply:
If the metrics is derived, we need to get all the datasets from the expression and skip using the value under `dataset` field in the `metricConfig`. If the metrics is not derived, we still need to add the `dataset` field in the `metricConfig` into the list. I am not sure that I can remove either the if-block or else-block.

Reply:
Minor: No need to add "else" after "return".
if () {
return()
}
// The logic for else